src/proto/extension.rs

```diff
@@ -111,6 +199,21 @@ impl Decode for KeySpec {
     }
 }
 
+impl Encode for KeySpec {
+    fn encoded_len(&self) -> ssh_encoding::Result<usize> {
+        [self.keyblob.encoded_len_prefixed()?, 1u8.encoded_len()?].checked_sum()
+    }
+
+    fn encode(&self, writer: &mut impl Writer) -> ssh_encoding::Result<()> {
+        self.keyblob.encode_prefixed(writer)?;
+        if self.is_ca {
```

Comment:
Is that okay if I leave a TODO for when we upstream that to `RustCrypto/SSH`? it will be easier then.
(otherwise i have to make a backport and a release, and it's a bit inconvenient)

Reply:
Sure. This seems like the best solution. Thanks!